packages/starknet-snap/src/ui/utils.test.tsx

```diff
@@ -0,0 +1,101 @@
+import type { FormattedCallData } from '../types/snapState';
+import { DEFAULT_DECIMAL_PLACES } from '../utils/constants';
+import { accumulateTotals } from './utils';
+
+describe('calculateTotals', () => {
+  const mockCalls = (overrides = [{}]) =>
+    [
+      {
+        tokenTransferData: {
+          amount: '1000000000000000000', // 1 ETH as string BigInt
+          symbol: 'ETH',
+          decimals: 18,
+          ...overrides[0],
+        },
+      },
+      {
+        tokenTransferData: {
+          amount: '500000000000000000', // 0.5 ETH as string BigInt
+          symbol: 'ETH',
+          decimals: 18,
+          ...overrides[1],
+        },
+      },
+      {
+        tokenTransferData: {
+          amount: '2000000000000000000', // 2 STRK as string BigInt
+          symbol: 'STRK',
+          decimals: 18,
+          ...overrides[2],
+        },
+      },
+    ] as FormattedCallData[];
+
+  const mockMaxFee = '100000000000000000'; // 0.1 token fee
+
+  it('sums up transfer amounts for each token', () => {
+    const calls = mockCalls();
+    const selectedFeeToken = 'ETH';
+
+    const result = accumulateTotals(calls, mockMaxFee, selectedFeeToken);
+
+    expect(result).toStrictEqual({
+      ETH: {
+        amount: BigInt('1600000000000000000'), // 1 + 0.5 + 0.1 ETH
+        decimals: 18,
+      },
+      STRK: {
+        amount: BigInt('2000000000000000000'), // 2 STRK
+        decimals: 18,
+      },
+    });
+  });
+
+  it('handles fee-only transactions when there are no transfers', () => {
```

Comment:
i think you can combine this with handles no calls gracefully
```suggestion
  it('returns the TokenTotals for the fee token only if there is no transfer type callData', () => {
```